### PR TITLE
feat: use default elasticsearch indexing instead of SearchIndexTask

### PR DESCRIPTION
**Pull request by mussbach**
**Ticket:** https://yaits.demos-deutschland.de/T28243

SearchIndexTask was once introduced to improve performance. 
This PR disables the reindexing via SerchIndexTask as we want to test, whether the default
indexing via fos elastica listener works and does not lead to
performance issues. When it does, we can reenable this code

### How to review/test
Delete Statement with Segments. It should work now

### PR Checklist

Delete the checkbox if it doesn't apply/isn't necessary.

- [ ] Tests updated/created
- [ ] Update documentation
- [ ] Link all relevant tickets
- [ ] Move the tickets on the board accordingly

### CODE REVIEW

**Comment by martendemos** [comment]
Deleting these methods is not an option?
Do we want mark these methods as deprecated or something?

**Comment by mussbach** [comment]
Usually I like to delete code immediately. My rationale was that we may need to be very quick in case of an incident when we have to rollback this change. When just commenting out we could easily undo the change. After some battle testing in production we could delete a whole bunch of code

**Comment by mussbach** [comment]
Deprecation hints: 92215e5f

**Review by AlexMenschoid — COMMENTED** [review]
I'm pretty sure that this PR should point towards `main` branch to be tested during the next sprint. I also would rather we delete the code and really thoroughly test it during the sprint, maybe even in different projects, to be sure it works before deploying it. But saying we leave it in like this suggests to me a lack of confidence which means we need more time with it before deploying it.

File: demosplan/DemosPlanCoreBundle/Logic/Statement/StatementService.php [hunk 1]
```diff
@@ -2032,23 +1950,11 @@ public function addVote($statementId, User $user)
...
-            $vote = $this->statementVoteRepository->findOneBy([
+            return $this->statementVoteRepository->findOneBy([
                 'user'      => $user->getId(),
                 'statement' => $statementId,
                 'deleted'   => false,
                 'active'    => true, ]);
-
-            // user already voted this statement?
-            if (null === $vote) {
-                $vote = $this->statementVoteRepository->add($data);
```
Comment (dresslerdemos) [inline demosplan/DemosPlanCoreBundle/Logic/Statement/StatementService.php:2043]: `$this->statementVoteRepository->add($data);` really not needed anymore?

**Review by dresslerdemos — CHANGES_REQUESTED** [review]
I don't understand how updating the elasticsearch index works now, but if it does work, then the removal of the code was done properly except for the few things noted in this review.

Please verify that `isElasticsearchAsyncIndexing`, `isElasticsearchAsyncIndexingLogStatus` and `getElasticsearchAsyncIndexingPoolSize` are removed  from `GlobalConfigInterface`.

`ResourceChange::getEntityIdsToUpdateInIndex` (and related code) can be removed too. I'd suggest to do this in a separate PR, but please create a corresponding ticket.

File: demosplan/DemosPlanCoreBundle/Logic/Statement/StatementService.php (hunk 1 above)
Comment (mussbach) [reply demosplan/DemosPlanCoreBundle/Logic/Statement/StatementService.php:2043]: Good catch, thx! f9c4a333

**Comment by mussbach** [comment]
Indexing works now via default fos elastica listeners defined by `listener: ~` in fos_elastica.yaml. The methods are removed from `GlobalConfigInterface`

**Comment by martendemos** [comment]
soll auch noch in den release_ewm branch

**Pull request merged**